src/sentry/models/environment.py

```diff
@@ -88,11 +88,16 @@ def get_or_create(cls, project, name):
         return env
 
     def add_project(self, project):
-        try:
-            with transaction.atomic():
-                EnvironmentProject.objects.create(project=project, environment=self)
-        except IntegrityError:
-            pass
+        cache_key = 'envproj:c:%s:%s' % (self.id, project.id)
+
+        if cache.get(cache_key) is None:
+            try:
+                with transaction.atomic():
+                    EnvironmentProject.objects.create(project=project, environment=self)
+                cache.set(cache_key, 1, 3600)
```

Comment:
Do we need to be explicit with the TTL here? I thought that `django.cache` would use the backend timeout by default.

Reply:
We generally favor being overly explicit.

Reply:
:+1: